rpcclient/zmq.go

```diff
@@ -0,0 +1,39 @@
+package rpcclient
+
+import (
+	"encoding/json"
+
+	"github.com/btcsuite/btcd/btcjson"
+)
+
+// FutureGetZmqNotificationsResult is a future promise to deliver the result of
+// a GetZmqNotifications RPC invocation
+type FutureGetZmqNotificationsResult chan *response
+
+// Receive waits for the response promised by the future and returns the unmarshalled
+// response, or an error if the request was unsuccessful.
+func (r FutureGetZmqNotificationsResult) Receive() ([]btcjson.ZmqNotificationResult, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+	var notifications []btcjson.ZmqNotificationResult
```

Comment:
I think it's better if `btcjson.GetZmqNotificationsResult` models _exactly_ the response of the RPC server, since someone may want to just use `btcjson` outside the context of btcd's rpcclient. This means the result type (currently in `btcjson/chainsvrresults.go`) should be defined as:

```go
type GetZmqNotificationsResult []struct {
    ...
}
```

as opposed to

```go
type GetZmqNotificationResult struct {
    ...
}
```